vendor/knative.dev/pkg/test/helpers/name.go

```diff
@@ -74,7 +76,7 @@ func AppendRandomString(prefix string) string {
 func RandomString() string {
 	suffix := make([]byte, randSuffixLen)
 	for i := range suffix {
-		suffix[i] = letterBytes[rand.Intn(len(letterBytes))]
+		suffix[i] = letterBytes[nameRand.Intn(len(letterBytes))]
```

Comment:
This is create a race condition, it seems not safe to access in parallel see failure [here](https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_serving/15353/istio-latest-no-mesh_serving_main/1805776384606343168). cc @dprotaso

Reply:
We can fix either with a mutex or via implementing a locked source as standard lib does internally.